src/Bicep.Core/Registry/Providers/TypesV1Archive.cs

```diff
@@ -23,6 +23,14 @@ public static async Task<BinaryData> GenerateProviderTarStream(IFileSystem fileS
             await AddFileToTar(tarWriter, "index.json", indexJson);
 
             var indexJsonParentPath = Path.GetDirectoryName(indexJsonPath);
+            var rootTypesPath = indexJsonParentPath != null ? Path.Combine(indexJsonParentPath, "types.json") : null;
+
+            if (fileSystem.File.Exists(rootTypesPath))
+            {
+                var rootTypesJson = await fileSystem.File.ReadAllTextAsync(rootTypesPath);
+                await AddFileToTar(tarWriter, "types.json", rootTypesJson);
+            }
```

Comment:
Rather than hardcoding expectations about the path structure, could you update `GetAllUniqueTypePaths` to detect the correct path?

Reply:
Ahh I actually don't think this is needed anymore since my bicep-types changes got checked in to add the fallbackType & configType. Since we are getting distinct references that are in the index.json.